crates/evm/evm/src/executors/invariant/mod.rs

```diff
@@ -348,66 +307,45 @@ impl<'a> InvariantExecutor<'a> {
 
             // We stop the run immediately if we have reverted, and `fail_on_revert` is set.
             if self.config.fail_on_revert && invariant_test.reverts() > 0 {
-                return Err(TestCaseError::fail("call reverted"))
+                return Err(TestCaseError::fail("Revert occurred."))
             }
 
             while current_run.depth < self.config.depth {
-                // Check if the timeout has been reached.
-                if timer.is_timed_out() {
-                    // Since we never record a revert here the test is still considered
-                    // successful even though it timed out. We *want*
-                    // this behavior for now, so that's ok, but
-                    // future developers should be aware of this.
-                    return Err(TestCaseError::fail(TEST_TIMEOUT));
-                }
-
                 let tx = current_run.inputs.last().ok_or_else(|| {
-                    TestCaseError::fail("no input generated to called fuzz target")
+                    TestCaseError::fail("No input generated to call fuzzed target.")
                 })?;
-
-                // Execute call from the randomly generated sequence without committing state.
-                // State is committed only if call is not a magic assume.
-                let mut call_result = current_run
+                if current_run.executor.get_balance(tx.sender)? < tx.value {
```

Comment:
@QiuhaoLi I am looking into integrate this approach but I am not sure we should inflate sender's balance with the fuzzed value (nor restore the balance after the call) as it could result in false positives like in forked tests / token transfers, etc. Instead we could bound it in (0, current sender balance) but even so for long running campaigns the balance could get spent quickly and then the fuzzed calls will be performed with call value `U256::ZERO`

@0xalpharush any thoughts re to deal with this? thank you!